Backport of upstream glib commit 9e320e1c43a4770ed1532248fe5416eb0c618120
to the glib bundled inside pkg-config 0.29.2. C99 aliases `bool` to
`_Bool` and C23 makes `bool` a reserved keyword, so gcc 15 (default
-std=gnu23) refuses to compile the `gboolean bool;` field declaration
in goption.c. Rename the field to `boolean`.

Origin: https://lists.openembedded.org/g/openembedded-core/message/198131
Signed-off-by: Martin Jansa <martin.jansa@gmail.com>

diff --git a/glib/glib/goption.c b/glib/glib/goption.c
--- a/glib/glib/goption.c
+++ b/glib/glib/goption.c
@@ -166,7 +166,7 @@ typedef struct
   gpointer arg_data;
   union
   {
-    gboolean bool;
+    gboolean boolean;
     gint integer;
     gchar *str;
     gchar **array;
@@ -1600,7 +1600,7 @@ free_changes_list (GOptionContext *context,
           switch (change->arg_type)
             {
             case G_OPTION_ARG_NONE:
-              *(gboolean *)change->arg_data = change->prev.bool;
+              *(gboolean *)change->arg_data = change->prev.boolean;
               break;
             case G_OPTION_ARG_INT:
               *(gint *)change->arg_data = change->prev.integer;
